skins/laika/src/pages/membership_application.ts

```diff
@@ -27,67 +37,82 @@ interface MembershipAmountModel {
 	urls: any,
 	showMembershipTypeOption: Boolean,
 	initialFormValues: InitialMembershipData,
+	userDataKey: string
 }
 
 const pageData = new PageDataInitializer<MembershipAmountModel>( '#app' );
+
+const dataPersister = createDataPersister(
+	new LocalStorageRepository(),
+	FORM_NAMESPACE,
+	pageData.applicationVars.userDataKey
+);
+
 const i18n = new VueI18n( {
 	locale: DEFAULT_LOCALE,
 	messages: {
 		[ DEFAULT_LOCALE ]: pageData.messages,
 	},
 } );
 
-const store = createStore();
+const store = createStore( [
+	dataPersister.getPlugin( persistenceItems ),
+] );
 
-function initializePage(): void {
-	new Vue( {
-		store,
-		i18n,
-		render: h => h( App, {
-			props: {
-				assetsPath: pageData.assetsPath,
-				pageIdentifier: PAGE_IDENTIFIER,
-			},
-		},
-		[
-			h( Component, {
-				props: {
-					validateAddressUrl: pageData.applicationVars.urls.validateAddress,
-					validateEmailUrl: pageData.applicationVars.urls.validateEmail,
-					validateFeeUrl: pageData.applicationVars.urls.validateMembershipFee,
-					validateBankDataUrl: pageData.applicationVars.urls.validateIban,
-					validateLegacyBankDataUrl: pageData.applicationVars.urls.convertBankData,
-					paymentAmounts: pageData.applicationVars.presetAmounts.map( a => Number( a ) * 100 ),
-					countries: pageData.applicationVars.countries,
-					showMembershipTypeOption: pageData.applicationVars.showMembershipTypeOption,
-					paymentIntervals: pageData.applicationVars.paymentIntervals,
-				},
-			} ),
-			h( Sidebar, {
-				slot: 'sidebar',
-			} ),
-		] ),
-	} ).$mount( '#app' );
-}
+dataPersister.decryptInitialValues( persistenceItems ).then( () => {
+	let initialFormValues = null;
+	let initialBankAccountData = null;
 
-// The PHP serialization sends the initial for data as empty array (instead of empty object) when donation was anonymous
-if ( pageData.applicationVars.initialFormValues !== undefined && !( ( pageData.applicationVars.initialFormValues as any ) instanceof Array ) ) {
-	const initializationCalls = [
-		store.dispatch( action( NS_MEMBERSHIP_ADDRESS, initializeAddress ), pageData.applicationVars.initialFormValues ),
-	];
-	if ( pageData.applicationVars.initialFormValues.iban ) {
-		initializationCalls.push(
-			store.dispatch(
-				action( NS_BANKDATA, initializeBankData ),
-				{
-					accountId: pageData.applicationVars.initialFormValues.iban,
-					bankId: pageData.applicationVars.initialFormValues.bic || '',
-					bankName: pageData.applicationVars.initialFormValues.bankname || '',
-				}
-			)
-		);
+	if ( pageData.applicationVars.initialFormValues !== undefined ) {
```

Comment:
You lost the check for array here (see comment in the original code). It does not matter for the functionality, since your implementation in the dataInitalizer can handle arrays (Kudos for that, I liked that very much). But from a type(script) perspective, we're sloppy here.

I'd recommend keeping the comment that explains the situation.